parsing/dialects/intermediate/dialects/dialect1/src/test/java/org/apache/calcite/test/Dialect1ParserTest.java

```diff
@@ -4087,4 +4087,68 @@ private void checkExNotNull(SqlParserUtil.StringAndPos sap, Throwable thrown) {
         + "TIME ZONE 'gmt'";
     sql(sql).ok(expected);
   }
+
+  @Test public void testIfStmtBase() {
+    final String sql = "create procedure foo (bee integer) "
+        + "if bee = 2 then select bar; end if";
+    final String expected =
+        "CREATE PROCEDURE `FOO` (IN `BEE` INTEGER) "
+            + "IF (`BEE` = 2) THEN SELECT `BAR`;\n"
+            + "END IF";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testIfStmtMultipleStatements() {
+    final String sql = "create procedure foo (bee integer) "
+        + "if bee = 2 then select bar; select baz; end if";
+    final String expected =
+        "CREATE PROCEDURE `FOO` (IN `BEE` INTEGER) "
+            + "IF (`BEE` = 2) THEN SELECT `BAR`;\n"
+            + "SELECT `BAZ`;\n"
+            + "END IF";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testIfStmtWithElseIf() {
+    final String sql = "create procedure foo (bee integer) "
+        + "if bee = 2 then select bar; "
+        + "else if bee = 3 then select baz; "
+        + "end if";
+    final String expected =
+        "CREATE PROCEDURE `FOO` (IN `BEE` INTEGER) "
+            + "IF (`BEE` = 2) THEN SELECT `BAR`;\n"
+            + "ELSE IF (`BEE` = 3) THEN SELECT `BAZ`;\n"
+            + "END IF";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testIfStmtWithElseIfWithElse() {
+    final String sql = "create procedure foo (bee integer) "
+        + "if bee = 2 then select bar; "
+        + "else if bee = 3 then select baz; "
+        + "else select xyz;"
+        + "end if";
+    final String expected =
+        "CREATE PROCEDURE `FOO` (IN `BEE` INTEGER) "
+            + "IF (`BEE` = 2) THEN SELECT `BAR`;\n"
+            + "ELSE IF (`BEE` = 3) THEN SELECT `BAZ`;\n"
+            + "ELSE SELECT `XYZ`;\n"
+            + "END IF";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testIfStmtNestedIf() {
+    final String sql = "create procedure foo (bee integer, abc integer) "
+        + "if bee = 2 then "
+        + "select bar; "
+        + "if abc = 3 then select baz; end if;"
+        + "end if";
+    final String expected =
+        "CREATE PROCEDURE `FOO` (IN `BEE` INTEGER, IN `ABC` INTEGER) "
+            + "IF (`BEE` = 2) THEN SELECT `BAR`;\n"
+            + "IF (`ABC` = 3) THEN SELECT `BAZ`;\n"
+            + "END IF;\n"
+            + "END IF";
+    sql(sql).ok(expected);
+  }
```

Comment:
It would probably be useful to add a few failing test cases. For example, you cannot start an IF statement with an ELSE or an ELSE IF, and you cannot have an empty statement body (ex. "IF A>2 THEN END IF").

Reply:
Will add failing test soon.

Reply:
Failing test cases added in https://github.com/googleinterns/calcite/pull/245/commits/ff9b2f8abfc814cbb8815c38bb8a55b35e5da747